packages/core/src/compiler/Substance.ts

```diff
@@ -550,13 +549,96 @@ const checkField = (decons: Deconstructor, env: Env): ResultWithType => {
 };
 
 export const checkVar = (variable: Identifier, env: Env): ResultWithType => {
-  const type = env.vars.get(variable.value);
+  const type = env.vars.find((_, key) => key === variable.value);
   if (type) {
     return ok([type, env]);
   } else {
-    const [...possibleVars] = env.vars.keys();
+    const possibleVars = env.varIDs;
     // TODO: find vars of the same type for error reporting (need to check expr first)
     return err(varNotFound(variable, possibleVars));
   }
 };
 //#endregion
+
+//#region Substance pretty printer
+
+export const prettySubstance = (prog: SubProg): string =>
```

Comment:
might be nice to pull all the AST pretty printing code (Sub + Sty + DSL) into one module, as currently my code is scattered around a few different files

Reply:
Normally the pretty-printer ships with the compiler. I think it's okay to just export top-level functions in each compiler module (i.e. `Substance`, `Domain`, and `Style`).

Reply:
Sounds good to me!